lib/ansible/plugins/inventory/azure_rm.py

```diff
@@ -514,8 +524,9 @@ def _on_instanceview_response(self, vm_instanceview_model):
                                  for s in vm_instanceview_model.get('statuses', []) if self._powerstate_regex.match(s.get('code', ''))), 'unknown')
 
     def _on_nic_response(self, nic_model, is_primary=False):
-        nic = AzureNic(nic_model=nic_model, inventory_client=self._inventory_client, is_primary=is_primary)
-        self.nics.append(nic)
+        if 'type' in nic_model and nic_model['type'] == 'Microsoft.Network/networkInterfaces':
```

Comment:
Can you add a comment for the `else` scenario?
The `if` conditional check can be written as `if nic_model.get('type') == 'Microsoft.Network/networkInterfaces'`

Reply:
resolved